packages/skills/carpark_detection/strategy.py

```diff
@@ -60,10 +61,19 @@ def __init__(self, **kwargs) -> None:
         self.data_price_fet = kwargs.pop('data_price_fet') if 'data_price_fet' in kwargs.keys() else DEFAULT_PRICE
         super().__init__(**kwargs)
 
-        self.db = DetectionDatabase(db_dir)
-        self.data_price_fet = 2000
+        balance = self.context.ledger_apis.token_balance('fetchai', cast(str, self.context.agent_addresses.get('fetchai')))
+
+        if not os.path.isdir(db_dir):
+            print("WARNING - DATABASE dir does not exist")
+
+        self.db = DetectionDatabase(db_dir, False)
         self.lat = 43
         self.lon = 42
+        self.record_balance(balance)
+
+    def record_balance(self, balance):
```

Comment:
Why are you keeping track of balance and not just query it from the ledger when needed?

Reply:
This is not really to keep track of it, but to put it in the database so it can be read and displayed by the GUI.